src/test/java/org/opentripplanner/framework/collection/ListUtilsTest.java

```diff
@@ -12,4 +12,21 @@ void combine() {
     var combined = ListUtils.combine(List.of(1, 2, 3), List.of(5, 6, 7));
     assertEquals(List.of(1, 2, 3, 5, 6, 7), combined);
   }
+
+  private static String makeHello() {
+    return new String("HELLO");
+  }
+
+  @Test
+  void distinctByKey() {
+    var first = new Wrapper(10, makeHello());
+    var last = new Wrapper(20, "HI");
+    var duplicates = List.of(first, new Wrapper(20, makeHello()), last);
+
+    var deduplicated = ListUtils.distinctByKey(duplicates, w -> w.string);
+
+    assertEquals(List.of(first, last), deduplicated);
+  }
```

Comment:
I find it slightly difficult to understand what this test does.

Reply:
I've tried to improve the readability of the test and added comments explaining what is being tested.